### PR TITLE
iGEMM: fixed xdlops gemm with k reduction and k pack

### PR DESCRIPTION
**Pull request by zjing14**
- Fixed issues of determining k reduction
- Fixed issues of kpack for fp32
- Added float2 and float4 support into emulation path

### CODE REVIEW

File: src/kernels/composable_kernel/include/tensor_operation/xdlops_gemm.hpp [hunk 1]
```diff
@@ -885,16 +911,17 @@ struct XdlopsGemm_t
             auto pa = reinterpret_cast<const data_type*>(&a);
             auto pb = reinterpret_cast<const data_type*>(&b);
 
-            constexpr index_t nxdlops =
-                (sizeof(FloatA) * mfma_type.num_input_blks) / (mfma_type.k * sizeof(data_type));
-
 #if WORKAROUND_SWDEV_229564
```
Comment (asroy) [inline src/kernels/composable_kernel/include/tensor_operation/xdlops_gemm.hpp:914]: Please move the definition of WORKAROUND_SWDEV_229564 from this file to config.hpp. So all marcros are defined there

**Comment by TejashShah** [comment]
@zjing14 Does it improve performance or fix some bug? If bug, could you please explain what we are fixing in this PR?

**Comment by zjing14** [comment]
@TejashShah As the description said, the PR is fixing a bug in xdlops gemm for determining if the mfma is k reduction-based, and fixing a bug in kpack support for fp32.

**Comment by TejashShah** [comment]
> 
> 
> @TejashShah As the description said, the PR is fixing a bug in xdlops gemm for determining if the mfma is k reduction-based, and fixing a bug in kpack support for fp32.

Good to know that you were fixing the kpack support in fp32 as I didn't see any issue with fp16/bfp16.

File: src/kernels/composable_kernel/include/tensor_operation/xdlops_gemm.hpp (hunk 1 above)
Comment (atamazov) [reply src/kernels/composable_kernel/include/tensor_operation/xdlops_gemm.hpp:914]: Not all.

Comment (atamazov) [reply src/kernels/composable_kernel/include/tensor_operation/xdlops_gemm.hpp:914]: The definitions should have as small scope as possible.

**Comment by atamazov** [comment]
@zjing14 
>     * Fixed issues of determining k reduction
>     * Fixed issues of kpack for fp32

Could you please describe possible side effects of these issues? For example, is it possible to get GPU segfault?

**Pull request closed without merging**